woocommerce-gateway-stripe.php

```diff
@@ -331,11 +366,106 @@ public function filter_gateway_order_admin( $sections ) {
 			 */
 			public function gateway_settings_update( $settings, $old_settings ) {
 				if ( false === $old_settings ) {
-					$gateway  = new WC_Gateway_Stripe();
-					$fields   = $gateway->get_form_fields();
-					$defaults = array_merge( array_fill_keys( array_keys( $fields ), '' ), wp_list_pluck( $fields, 'default' ) );
-					return array_merge( $defaults, $settings );
+					$gateway      = new WC_Gateway_Stripe();
+					$fields       = $gateway->get_form_fields();
+					$old_settings = array_merge( array_fill_keys( array_keys( $fields ), '' ), wp_list_pluck( $fields, 'default' ) );
+					$settings     = array_merge( $old_settings, $settings );
```

Comment:
Shouldn't we `return $settings` here? I see we had an early return here in the previous implementation 🤔

Reply:
If the UPE feature flag is not enabled, yes.

See change [here](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1685/files#diff-e1e5e9b442c2b5d4105edbdc3787651333c1923dd623677025f72d25796900d6R379-R381).